internal/services/cdn/cdn_frontdoor_helpers.go

```diff
@@ -200,10 +208,31 @@ func expandStringSliceToCsvFormat(input []interface{}) *string {
 	return &csv
 }
 
+func expandCustomDomainActivatedResourceArray(input []interface{}) *[]cdn.ActivatedResourceReference {
+	results := make([]cdn.ActivatedResourceReference, 0)
+
+	// NOTE: I have confirmed with the service team that this is required to be an explicit "nil" value, an empty
+	// list will not work. I had to modify the SDK to allow for nil which in the API means disassociate the custom domains.
+	if len(input) == 0 || input == nil {
+		return nil
+	}
+
+	// Normalize these values, if these are imported from portal the will all be lowercased...
+	for _, customDomain := range input {
+		if id, err := parse.FrontDoorCustomDomainIDInsensitively(customDomain.(string)); err == nil {
```

Comment:
the expand function is coming from the config, so the casing on these will be validated correctly

```suggestion
		if id, err := parse.FrontDoorCustomDomainID(customDomain.(string)); err == nil {
```

should we also be raising this error?